base/src/org/compiere/model/MLandedCostAllocation.java

```diff
@@ -244,28 +247,26 @@ public int getC_DocType_ID() {
 	}
 
 	public BigDecimal getPriceActualCurrency() {
-		MInvoiceLine invoiceLine = (MInvoiceLine) getC_InvoiceLine();
-		MCurrency currency = MCurrency.get(getCtx(), getC_Currency_ID());
-		BigDecimal amount = getAmt().divide(getQty() , currency.getCostingPrecision() ,  RoundingMode.HALF_UP);
-		if (MDocType.DOCBASETYPE_APCreditMemo.equals(invoiceLine.getParent().getC_DocType().getDocBaseType()))
-			amount = amount.negate();
+		BigDecimal amount = getAmt(true, false).divide(getQty() , MathContext.DECIMAL128);
 		return  amount;
 	}
 
 	@Override
 	public int getC_Currency_ID ()
 	{
-		return DB.getSQLValue(get_TrxName() ,
-				"SELECT i.C_Currency_ID FROM C_InvoiceLine il INNER JOIN C_Invoice i ON (il.C_Invoice_ID=i.C_Invoice_ID) WHERE il.C_InvoiceLine_ID = ? ",
-				getC_InvoiceLine_ID());
+		if (getInvoice()!=null)
+			return getInvoice().getC_Currency_ID();

```

Comment:
Why use getInvoice().getC_Currency_ID() when this approach need load all PO object and only need a field , is inefficient and slow ?